### PR TITLE
files: add file permissions configurations

### PR DESCRIPTION
**Pull request by jma**
* Adds record files service permission configurations.
* Adds invenio record files support for the permission web API.
* Avoids document deletion if files exists.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

### CODE REVIEW

File: rero_ils/modules/files/permissions.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (lauren-d) [inline rero_ils/modules/files/permissions.py:4]: `Copyright (C) 2019-2024 RERO`

File: rero_ils/modules/files/services.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (lauren-d) [inline rero_ils/modules/files/services.py:4]: `Copyright (C) 2019-2024 RERO`

File: tests/api/files/test_files_permissions.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2020 RERO
+# Copyright (C) 2020 UCLouvain
```
Comment (lauren-d) [inline tests/api/files/test_files_permissions.py:5]: `Copyright (C) 2019-2024 RERO`

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/documents/api.py [hunk 4]
```diff
@@ -329,6 +331,7 @@ def get_links_to_me(self, get_pids=False):
         if get_pids:
             holdings = sorted_pids(hold_query)
             items = sorted_pids(item_query)
+            files = [hit.id for hit in file_query.scan()]
```
Comment (rerowep) [inline rero_ils/modules/documents/api.py:334]: Maybe use `.source('id')`.

File: rero_ils/modules/files/cli.py [hunk 5]
```diff
@@ -58,7 +58,7 @@ def create_pdf_file(document):
     return generator.output()
 
 
-def create_pdf_record_files(document, metadata, flush=False):
+def create_pdf_record_files(document, metadata, flush=False, n=None):
```
Comment (rerowep) [inline rero_ils/modules/files/cli.py:61]: Could you give more information for `n`. Minimum recommendation for variable/parameter length is 3! Missing in doc string `flush`, `n`.

File: rero_ils/modules/files/cli.py [hunk 6]
```diff
@@ -100,18 +111,18 @@ def create_files(number):
     doc_pids = list(Document.get_all_pids())
     lib_pids = list(Library.get_all_pids())
     shuffle(doc_pids)
```
Comment (rerowep) [inline rero_ils/modules/files/cli.py:113]: Why we need `shuffle` and why we have `collections = ["col1", "col2", "col3"]`?
Maybe use:

```python
@click.option('-c', '--collections', multiple=True, default=["col1", "col2", "col3"])
def create_files(number, collections):
```

**Review by rerowep — CHANGES_REQUESTED** [review]

**Comment by PascalRepond** [comment]
## PO tests

- [ ] REP: If I upload a file with an existing file name in the same bucket, the thumbnail is not displayed correctly.
image
- [ ] REP: When uploading a file, a warning should be displayed to the librarian “The files uploaded here will be publicly available. Make sure that you are legally allowed to distribute a file before adding it to a document.”
- [ ] REP: Collection list editor
    - [ ] Label should be "Digital collections"
    - [ ] The tooltip is in a different style/font than in the editors, should be the same
- [ ] REP: PRO UI: "My files" tab
    - [ ] I would rename the tab label to "Files" and adjust the title inside to "Files (my library)", if possible
    - [ ] The files in the list should be unextended by default
    - [ ] The button to extend could be clearer (an arrow on the left, like in the rest of the app instead of the "i")
    - [ ] There should be a button at the top to extend all elements at once

**Review by rerowep — APPROVED** [review]

**Pull request merged**